swarm/shed/index.go

```diff
@@ -262,3 +275,226 @@ func (f Index) IterateFrom(start IndexItem, fn IndexIterFunc) (err error) {
 	}
 	return it.Error()
 }
+
+// Subscription provides methods to control
+// and get information about subscription state.
+type Subscription struct {
+	stopChan chan struct{}
```

Comment:
why need noth this or why need context,

Reply:
Explanation https://github.com/ethersphere/go-ethereum/pull/1014#discussion_r236068738.